scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -177,11 +178,18 @@ static struct dom_ctx *lookup_dom_ctx(u32 dom_id)
 	return domc;
 }
 
+static int get_task_from_id(u64 task_id, struct task_struct *p_in){
+	struct task_struct *p;
+	p_in = (struct task_struct*) task_id;
```

Comment:
Does this work? I'm not sure BPF would allow casting from u64 to a pointer like this. One way to obtain the trusted `task_struct` from the ID would be doing `probe_read` on the `&((struct task_struct *)task_id)->pid` and then do `bpf_task_from_pid()` on the read PID.

Reply:
Also, wouldn't a more conventional name for the out parameter be `p_out` instead of `p_in`?